content/ja/handbook/engineering/ai/ai-coding/code_suggestions/development_guide/implementation_guidelines.md

```
@@ -0,0 +1,91 @@
+
+- モデルへのリクエストで **stop tokens** を指定したことを確認してください。([MR の例](https://gitlab.com/gitlab-org/modelops/applied-ml/code-suggestions/ai-assist/-/merge_requests/1298) を参照。)
+- モデルとユースケースに応じて、リクエストに **追加のコンテキストを含める** 必要がある場合があります。これには、プロジェクトライブラリなどの開いているファイルの内容を追加することが含まれます。
+- ほとんどのモデルでは、空白のトリミングやコメントの削除などの **後処理** が必要です。既存の post-processor を使用するか ([MR の例](https://gitlab.com/gitlab-org/modelops/applied-ml/code-suggestions/ai-assist/-/merge_requests/1351/))、必要に応じて新しいものを追加できます ([MR の例](https://gitlab.com/gitlab-org/modelops/applied-ml/code-suggestions/ai-assist/-/merge_requests/1470))。
+
+## GitLab Rails
+
+[GitLab Rails リポジトリ](https://gitlab.com/gitlab-org/gitlab/)。
+
+GitLab Rails は AI モデルにリクエストを直接送信 _しません_。AIGW にリクエストを送信し、AIGW が AI モデルにリクエストを送信します。
+
+GitLab Rails は、Code Completions または Code Generations にどのモデルを使用するかについての唯一の信頼できる情報源です。[機能フラグ](#introduce-behind-a-feature-flag) を介して現在のモデルを切り替えます。
```

markdownlint-cli2 (0.22.1)

[warning] Link fragments should be valid

(MD051, link-fragments)